tests/system/providers/google/cloud/sql_to_sheets/example_sql_to_sheets.py

```diff
@@ -66,18 +60,72 @@
 """
 from __future__ import annotations
 
+import json
+import logging
 import os
 from datetime import datetime
 
+from googleapiclient import discovery
+
 from airflow import models
+from airflow.decorators import task
+from airflow.models import Connection
+from airflow.operators.bash import BashOperator
+from airflow.providers.common.sql.operators.sql import SQLExecuteQueryOperator
+from airflow.providers.google.cloud.operators.cloud_sql import (
+    CloudSQLCreateInstanceDatabaseOperator,
+    CloudSQLCreateInstanceOperator,
+    CloudSQLDeleteInstanceOperator,
+)
+from airflow.providers.google.suite.operators.sheets import GoogleSheetsCreateSpreadsheetOperator
 from airflow.providers.google.suite.transfers.sql_to_sheets import SQLToGoogleSheetsOperator
-
-ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
-DB_CONNECTION = os.environ.get("DB_CONNECTION")
-SPREADSHEET_ID = os.environ.get("SPREADSHEET_ID", "test-id")
+from airflow.settings import Session
+from airflow.utils.trigger_rule import TriggerRule
 
 DAG_ID = "example_sql_to_sheets"
-SQL = "select col2 from test_table"
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID", "default")
+PROJECT_ID = os.environ.get("SYSTEM_TESTS_GCP_PROJECT", "default")
+
+SHEETS_CONNECTION_ID = f"connection_{DAG_ID}_{ENV_ID}"
+SPREADSHEET = {
+    "properties": {"title": "Test1"},
+    "sheets": [{"properties": {"title": "Sheet1"}}],
+}
+
+CLOUD_SQL_INSTANCE = f"cloud-sql-{DAG_ID}-{ENV_ID}".replace("_", "-")
+CLOUD_SQL_INSTANCE_CREATION_BODY = {
+    "name": CLOUD_SQL_INSTANCE,
+    "settings": {
+        "tier": "db-custom-1-3840",
+        "dataDiskSizeGb": 30,
+        "ipConfiguration": {
+            "ipv4Enabled": True,
+            "requireSsl": False,
+            # Consider specifying your network mask
+            # for allowing requests only from the trusted sources, not from anywhere
+            "authorizedNetworks": [
+                {"value": "0.0.0.0/0"},
```

Comment:
I'm afraid this approach with CloudSQL won't work in our CI projects because authorized access is restricted there. Thus you could use another approach based on GCE+docker: https://github.com/apache/airflow/pull/34502/files#diff-7ba589634412647aa0df2a959d39d754410a7efe4e2b85cf5c958c5f21c461ff